crates/ruma-macros/src/events/event_content.rs

```diff
@@ -579,75 +548,55 @@ fn generate_event_content_without_relation<'a>(
     );
     let without_relation_ident = format_ident!("{ident}WithoutRelation");
 
-    match without_relation_type {
-        WithoutRelationType::Generated => {
-            let with_relation_fn_doc =
-                format!("Transform `self` into a [`{ident}`] with the given relation.");
-
-            let (relates_to, other_fields) = fields.partition::<Vec<_>, _>(|f| {
-                f.ident.as_ref().filter(|ident| *ident == "relates_to").is_some()
-            });
+    let with_relation_fn_doc =
+        format!("Transform `self` into a [`{ident}`] with the given relation.");
 
-            let relates_to_type =
-                relates_to.into_iter().next().map(|f| &f.ty).ok_or_else(|| {
-                    syn::Error::new(
-                        Span::call_site(),
-                        "`without_relation` can only be used on events with a `relates_to` field",
-                    )
-                })?;
+    let (relates_to, other_fields) = fields.partition::<Vec<_>, _>(|f| {
+        f.ident.as_ref().filter(|ident| *ident == "relates_to").is_some()
+    });
 
-            let without_relation_fields =
-                other_fields.iter().flat_map(|f| &f.ident).collect::<Vec<_>>();
-            let without_relation_struct = if other_fields.is_empty() {
-                quote! { ; }
-            } else {
-                quote! {
-                    { #( #other_fields, )* }
-                }
-            };
+    let relates_to_type = relates_to.into_iter().next().map(|f| &f.ty).ok_or_else(|| {
+        syn::Error::new(
+            Span::call_site(),
+            "`without_relation` can only be used on events with a `relates_to` field",
+        )
+    })?;
 
-            Ok(quote! {
-                #[automatically_derived]
-                impl #ruma_common::events::EventContentWithoutRelation for #ident {
-                    type WithoutRelation = #without_relation_ident;
-                }
+    let without_relation_fields = other_fields.iter().flat_map(|f| &f.ident).collect::<Vec<_>>();
+    let without_relation_struct = if other_fields.is_empty() {
+        quote! { ; }
```

Comment:
This isn't exactly right: This would "catch" both `struct SomeEventContent {}` and `struct SomeEventContent;`. I wonder whether we can refactor this `fn` to take `&syn::Fields` instead of an `IntoIterator<Item = &'a Field>` 🤔

Reply:
I copied this from the part that generates the redacted content. Should it also be fixed there?

Reply:
Hm, well there we actually filter the fields so Ibwonder what's the best behavior... Anyways, I'm probably being to nit-picky and we're probably consistently using unit structs instead of fieldless record structs.